scripts/metricsgen/metricsdiff/metricsdiff.go

```diff
@@ -0,0 +1,156 @@
+// metricsdiff is a tool for generating a diff between two different files containing
+// prometheus metrics. metricsdiff outputs which metrics have been added, removed,
+// or have different sets of labels between the two files.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `Usage: %[1]s <path> <path>
+
+Generate the diff between the sets of prometheus metrics peresent in the two files
+in the <path> files. The tool outputs which metrics have been added, removed, or
+use different label sets between the two files.
+
+`, filepath.Base(os.Args[0]))
+		flag.PrintDefaults()
+	}
+}
+
+// MetricsDiff contains the set of metrics that were modified between two files
+// containing prometheus metrics output.
+type MetricsDiff struct {
+	Adds    []string
+	Removes []string
+
+	LabelDiffs []LabelDiff
+}
+
+// MetricsDiff contains the set of labels that were modified for a single metric
+// between two files containing prometheus metrics output.
+type LabelDiff struct {
+	MetricsName string
+	Adds        []string
+	Removes     []string
+}
+
+func main() {
+	flag.Parse()
+	if len(os.Args) != 3 {
+		log.Fatalf("Incorrect number of arguments, expected 2 but received %d", len(os.Args)-1)
+	}
+	fa, err := os.Open(os.Args[1])
+	if err != nil {
+		log.Fatalf("Unable to open file %s: %v", os.Args[1], err)
+	}
+	fb, err := os.Open(os.Args[2])
+	if err != nil {
+		log.Fatalf("Unable to open file %s: %v", os.Args[2], err)
+	}
+	md, err := MetricsDiffFromReaders(fa, fb)
+	if err != nil {
+		log.Fatalf("Generating diff: %v", err)
+	}
+	fmt.Printf("%s", md)
```

Comment:
```suggestion
	fmt.Println(md)
```

or `fmt.Print(md)` if you don't want an extra newline.